modules/onewire/onewire.h

```diff
@@ -0,0 +1,287 @@
+/**
+ * |----------------------------------------------------------------------
+ * | Copyright (C) Tilen Majerle, 2014
+ * |
+ * | This program is free software: you can redistribute it and/or modify
+ * | it under the terms of the GNU General Public License as published by
+ * | the Free Software Foundation, either version 3 of the License, or
+ * | any later version.
+ * |
+ * | This program is distributed in the hope that it will be useful,
+ * | but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * | GNU General Public License for more details.
+ * |
+ * | You should have received a copy of the GNU General Public License
+ * | along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * |----------------------------------------------------------------------
+ */
+
+
+
+#ifndef ONEWIRE_H
+#define ONEWIRE_H
+
+/* C++ detection */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// includes to access HAL/PAL calls, need to access directly since abstraction for ChibiOS in framework is lackluster
+#include <hal.h>                   // all encompasing hal.h include. includes everything related to hal that you would need.
+#include <modules/timing/timing.h> // need timing module for microsecond level bit-banging

```

Comment:
Are these needed in the header file?  Can they be moved to onewire.c instead? (After a quick scan of your header file, I don't think these need to be in this file, but I may have missed something).

Reply:
No you are correct. I will move these.